funcs_scripts.php

```diff
@@ -254,6 +260,56 @@ function is_gha_repository()
     );
 }
 
+/**
+ * Determine if the module being processed is "TOOLING"
+ *
+ * Example usage:
+ * is_gha_repository()
+ */
+function is_tooling()
+{
+    global $GITHUB_REF;
+    return in_array(
+        $GITHUB_REF,
+        array_column(
+            MetaData::getAllRepositoryMetaData()[MetaData::CATEGORY_TOOLING],
+            'github'
+        )
+    );
+}
+
+/**
+ * Determine if the module being processed is "MISC"
+ *
+ * Example usage:
+ * is_gha_repository()
+ */
+function is_misc()
+{
+    global $GITHUB_REF;
+    return in_array(
+        $GITHUB_REF,
+        array_column(
+            MetaData::getAllRepositoryMetaData()[MetaData::CATEGORY_MISC],
+            'github'
+        )
+    );
+}
+
+/**
+ * Determine if the module being processed has a wildcard major version mapping
+ * in silverstripe/supported-modules repositories.json
+ *
+ * Example usage:
+ * has_wildcard_major_version_mapping()
+ */
+function has_wildcard_major_version_mapping()
+{
+    global $GITHUB_REF;
+    $repoData = MetaData::getMetaDataForRepository($GITHUB_REF);
+    return array_key_exists('*', $repoData['majorVersionMapping']);
```

Comment:
Do we specifically care about when there is a wildcard, or only when there *isn't* a numbered major version?
I don't think there's any right now with both, but the readme and unit tests in supported-modules repo allow for it in the future.

Reply:
When there's not a numbered major I guess? I don't quite understand the question though

A wildcard is specifically for when the module doesn't map to CMS majors? e.g. rhino

If there's a wildcard, kind of implies there is not and will never be a version that maps to a CMS major

Reply:
Regardless, the method does what its name implies

Reply:
I was going to drop this as not worth further ping pong, except the comment in the dispatch-ci script makes it clear this is important:
> If module non has_wildcard_major_version_mapping then dispatch-ci.yml should always be present

It is possible in the future we could have a scenario where (for example) old branches are for specific CMS versions, but the new branch is for *all* cms versions. Or where for some other reason we have both a wildcard entry *and* versions that map to CMS majors.

```json
{
"majorVersionMapping": {
        "4": ["2", "3"],
        "5": ["4"],
        "*": ["5"]
      }
}
```

As I mentioned earlier
> I don't think there's any right now with both, but the readme and unit tests in supported-modules repo allow for it in the future.

So it's important to know whether the intention here is to explicitly ignore CMS versions if there is any wildcard entry.

Reply:
> So it's important to know whether the intention here is to explicitly ignore CMS versions if there is any wildcard entry.

Yeah that's the intention. The assumption is that it's either wildcard = main/master style, and not wildcard = semver

> It is possible in the future we could have a scenario where (for example) old branches are for specific CMS versions, but the new branch is for all cms versions.

We're running a pretty strict policy of "new module major for new cms major", I wouldn't expect that to change anytime soon

Reply:
Have discussed offline and created PR to restrict wildcard https://github.com/silverstripe/supported-modules/pull/33